src/dfx/src/config/dfinity.rs

```diff
@@ -61,18 +64,73 @@ pub const DEFAULT_IC_GATEWAY_TRAILING_SLASH: &str = "https://ic0.app/";
 /// A Canister configuration in the dfx.json config file.
 /// It only contains a type; everything else should be infered using the
 /// CanisterInfo type.
-#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct ConfigCanistersCanister {
-    pub r#type: Option<String>,
-
     #[serde(default)]
     pub declarations: CanisterDeclarationsConfig,
 
     #[serde(default)]
     pub remote: Option<ConfigCanistersCanisterRemote>,
 
-    #[serde(flatten)]
-    pub extras: BTreeMap<String, Value>,
+    pub args: Option<String>,
+
+    #[serde(default)]
+    pub initialization_values: InitializationValues,
+
+    #[serde(default)]
+    pub dependencies: Vec<String>,
+
+    #[serde(default)]
+    pub frontend: bool,
```

Comment:
```suggestion
    #[serde(default)]
    pub is_frontend: bool,
```
That or a comment explaining what it does seems appropriate

Reply:
I'm trying to break absolutely nothing (see PossiblyStr) - we can evolve the config to make more sense in a future PR